### ISSUE TITLE
Bug: region gets reset

### ISSUE THREAD

**Issue by moltar** [common-fate/granted#528]
This is a really odd issue that I have never seen before...

---

I have SSO setup for `region-FOO`.

When I run:

```sh
assume --region=region-BAR
```

I get the right region in the console:

```
❯ assume --region=region-BAR

? Please select the profile you would like to assume: Foo/AdministratorAccess
[i] To assume this profile again later without needing to select it, run this command:
> assume Foo/AdministratorAccess --region=region-BAR

~ on ☁️  Foo/AdministratorAccess (region-BAR) took 3s
```

Then _any_ terminal command, even not related to AWS, will reset the region:

```sh
~ on ☁️  Foo/AdministratorAccess (region-BAR)
❯ ls
```

Now the region gets reset to the SSO configured one:

```sh
~ on ☁️  Foo/AdministratorAccess (region-FOO)
❯
```

---

Using:

```
zsh 5.9 (arm-apple-darwin22.1.0)
```

```
❯ starship --version
starship --version
starship 1.16.0
branch:
commit_hash:
build_time:2023-07-30 13:30:05 +00:00
build_env:rustc 1.71.0 (8ede3aae2 2023-07-12) (built from a source tarball),
```

**Comment by moltar** [comment]
Tested different versions, and this issues starts happening on `v0.19.0`.

**Comment by shwethaumashanker** [comment]
@moltar Could you also please send an example of a profile from `~/.aws/config` that has this issue? Feel free to remove account ID and other sensitive data

**Comment by moltar** [comment]
```ini
[profile AD/Foo/AdministratorAccess]
granted_sso_start_url      = https://abc.awsapps.com/start
granted_sso_region         = ca-central-1
granted_sso_account_id     = 1234
granted_sso_role_name      = AdministratorAccess
common_fate_generated_from = aws-sso
credential_process         = granted credential-process --profile AD/Foo/AdministratorAccess
```

**Comment by shwethaumashanker** [comment]
@moltar I'm having a hard time replicating the error. Can you please run `env | grep AWS` and check if the `AWS_REGION` is set correctly? Also, do you have the same issue with v0.19.1, [this might](https://github.com/common-fate/granted/pull/521) have fixed your issue

**Comment by moltar** [comment]
> Can you please run env | grep AWS and check if the AWS_REGION is set correctly?

That's the thing - any command will reset the region, so running `env` will do it too.

> Also, do you have the same issue with v0.19.1

Yes, happens on that version too.

**Comment by shwethaumashanker** [comment]
@moltar By any chance do you have [automatic reassume roles](https://docs.commonfate.io/granted/recipes/automatically_reassume/) enabled ? This runs every time you run a command in zsh and may be the cause of the problem

**Comment by moltar** [comment]
Yes, I do have that enabled. I assumed that was the issue, given that this was what was released or updated in later versions. But is that really the expected behaviour?

---

```
❯ cat ~/.granted/config
DefaultBrowser = "FIREFOX"
CustomBrowserPath = "/opt/homebrew/bin/firefox"
CustomSSOBrowserPath = ""
Ordering = ""
ExportCredentialSuffix = ""

[Keyring]
  Backend = "keychain"
```

**Comment by chrnorm** [comment]
Here's a quick TLDR of what I think the underlying issue here is, plus a longer explanation below.

TLDR: a change we made in #467 is causing the ZSH auto-reassume hook to trigger on every command:

```bash
assume --region us-west-1 my-profile

# then, immediately after, run another command:
ls
<- granted_auto_reassume zsh function is called immediately here
```

@moltar given that you are using the `credential_process` integration, could you please test switching off the [ZSH automatic reassumption hook](https://docs.commonfate.io/granted/recipes/automatically_reassume/) and let us know if your workflows still work? You can do this by removing the `export GRANTED_ENABLE_AUTO_REASSUME=true` line from your `~/.zshrc`.



## Longer explanation



I think the issue here is that #467 (which shipped in v0.19.0) has changed the behaviour so that the auto reassume hook [here](https://github.com/common-fate/granted/blob/cb88df1e9694aa5d108f7a12e690d2d2c59c0823/scripts/assume#L137) is being triggered on every command rather than when a session has expired. #467 changed the behaviour for profiles with a `credential_process` so that only `AWS_PROFILE` is exported, meaning the AWS CLI will automatically refresh session credentials without the need for any shell hooks. You can read about this in #263.

I think this is occurring because `AWS_SESSION_EXPIRATION` is no longer being set, so [this check](https://github.com/common-fate/granted/blob/cb88df1e9694aa5d108f7a12e690d2d2c59c0823/scripts/assume#L126) might fail, causing `assume` to be re-executed each time a shell command is run.



The change we made in #467 should negate the need to use the ZSH auto-reassumption hook altogether. We'll update our documentation to clarify this. I'll leave this issue open until we deal with the fact that the hook is being called repeatedly, and also that when the hook is called it doesn't respect the existing `AWS_REGION` environment variable.
